### PR TITLE
feat: update `app.{set|get}LoginItemSettings(settings)`

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Closes https://github.com/electron/electron/issues/37560.
Refs https://github.com/electron/electron/issues/37228
Refs https://github.com/electron/governance/pull/533

This PR updated `app.setLoginItemSettings(settings)` and `app.getLoginItemSettings(settings)` on macOS 13 to use a single streamlined underlying API on both Mac App Store and regular macOS builds. This will remove several configuration and settings options, including `openAsHidden` in `app.setLoginItemSettings` and `wasOpenedAtLogin` and `wasOpenedAsHidden` in `app.getLoginItemSettings`.

This simplifies the underlying logic in accordance to the APIs apple has outlined for usage and improve parity between MAS and non-MAS builds. This will also allow developers to register app service objects, launch daemons, and login item services to launch on startup.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation, tutorials, templates and examples are changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: Migrated `app.{set|get}LoginItemSettings(settings)` to use Apple's new recommended underlying framework on macOS.

### CODE REVIEW

File: docs/api/app.md
Comment (kyleshay) [inline docs/api/app.md:1283]: `wasOpenedAtLogin` is currently working for me on macOS 13.2 (only `wasOpenedAsHidden` no longer works) 

If that's the case, should it be mentioned to not rely on `openAtLogin` as well on L1281 (and below in the:
`### `app.setLoginItemSettings(settings)` _macOS_ _Windows_` section?)

File: docs/api/app.md
Comment (kyleshay) [inline docs/api/app.md:1286]: Is `status` only macOS 13 and up?

**Comment by quanglam2807** [comment]
Is this blocked by something, @codebytere?

File: docs/api/app.md
Comment (codebytere) [reply docs/api/app.md:1305]: The latter, yes. It's all due to be removed in favor of `SMAppService` APIs.

File: docs/api/app.md
Comment (erickzhao) [inline docs/api/app.md:1328]: Should we duplicate these `type` descriptions between this and the `app.getLoginItemSettings` `type` property on line 1293?

File: docs/api/app.md
Comment (erickzhao) [inline docs/api/app.md:1304]: Would it make sense to put these "This does not work on macOS 13 and up" settings through the formal deprecation cycle?

**Review by erickzhao — APPROVED** [review]
API LGTM

File: docs/api/app.md
Comment (codebytere) [reply docs/api/app.md:1304]: We could consider it, but it might be a bit tricky figuring out how and where to emit given it's gin-converted properties being deprecated. What would you imagine it look like? A deprecation warning per-property? Or simply once when the function is invoked with a comma-separated list of deprecated properties?

**Review by VerteDinde — APPROVED** [review]
LGTM - test failures are unrelated

**Review by jkleinsc — COMMENTED** [review]
API LGTM

File: shell/browser/api/electron_api_app.cc
Comment (jkleinsc) [inline shell/browser/api/electron_api_app.cc:364]: This assignment seems out of place.  Is there a reason for it here?

**Comment by codebytere** [comment]
Working on trying to add some tests for this before it's merged.

File: docs/api/app.md
Comment (codebytere) [reply docs/api/app.md]: noted in docs!

File: docs/api/app.md
Comment (codebytere) [reply docs/api/app.md:1316]: linked them

File: docs/api/app.md
Comment (jkleinsc) [inline docs/api/app.md:1324]: Looks like this property is now available on macOS, so it should be documented here.

File: docs/api/app.md
Comment (jkleinsc) [inline docs/api/app.md:1282]: `name` exists on Windows as well and should be documented here.

Comment (codebytere) [reply docs/api/app.md]: Addressed in [9ceb66e](https://github.com/electron/electron/pull/37244/commits/9ceb66eb74dfcbf80bc2e9445a27a7cbca095f57)

**Review by jkleinsc — APPROVED** [review]

**Pull request merged**
